src/meta/meta_ckpt.c

```diff
@@ -976,8 +983,27 @@ __wt_meta_sysinfo_set(WT_SESSION_IMPL *session)
         WT_ERR(__wt_metadata_update(session, WT_SYSTEM_OLDEST_URI, buf->data));
     }
 
+    /*
+     * Record snapshot information in metadata for checkpoint.
+     *
+     */
+    if (txn->snapshot_count > 0) {
+        WT_ERR(__wt_buf_fmt(session, buf,
+          WT_SYSTEM_CKPT_SNAPSHOT_MIN "=%" PRIu64 "," WT_SYSTEM_CKPT_SNAPSHOT_MAX "=%" PRIu64
+                                      "," WT_SYSTEM_CKPT_SNAPSHOT_COUNT "=%" PRIu32
+                                      "," WT_SYSTEM_CKPT_SNAPSHOT "=",
+          txn->snap_min, txn->snap_max, txn->snapshot_count));
+
+        for (i = 0; i < txn->snapshot_count; i++) {
+            WT_RET(__wt_snprintf(buf2, sizeof(buf2), "%" PRIu64, txn->snapshot[i]));
+            WT_ERR(__wt_buf_catfmt(session, buf, "%s%s", buf2, sep));
+        }
+        WT_ERR(__wt_metadata_update(session, WT_SYSTEM_CKPT_SNAPSHOT_URI, buf->data));
+    }
+
 err:
     __wt_scr_free(session, &buf);
+    __wt_free(session, buf2);
```

Comment:
buf2 is a local variable. No need to free it.

Reply:
Fixed